qemu/tests/balloon_uniqueness.py

```diff
@@ -0,0 +1,53 @@
+from virttest import env_process
+from virttest.qemu_devices.qdevices import QDevice
+
+from virttest.virt_vm import VMCreateError
+from virttest.qemu_monitor import QMPCmdError
+
+
+def run(test, params, env):
+    """
+    Boot guest with or hotplug more than 1 balloon devices, qemu should reject,
+    3 scenarios:
+    1. Boot guest with two balloons
+    2. Boot guest with one balloon and then hot plug the second one
+    3. Boot guest with none balloon and then hot plug 2 balloon devices
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    params['start_vm'] = 'yes'
+    vm_name = params['main_vm']
+    boot_warning_msg = params.get('boot_warning_msg')
+    try:
+        env_process.preprocess_vm(test, params, env, vm_name)
+    except VMCreateError as e:
+        if not boot_warning_msg or boot_warning_msg not in e.output:
+            raise
+    else:
+        if boot_warning_msg:
+            test.fail('The guest should not start with two balloon devices.')
+
+    machine_type = params['machine_type']
+    bus = {'aobject': 'pci.0'}
+    if 's390' in machine_type:  # For s390x platform
+        model = 'virtio-balloon-ccw'
+        bus = {'type': 'virtio-bus'}
+    else:
+        model = 'virtio-balloon-pci'
+    hotplug = int(params.get('hotplug', 0))
+    for i in range(hotplug):
+        dev = QDevice(model, parent_bus=bus)
+        dev.set_param('id', 'hotplugged_balloon%s' % i)
+        dev_num = len(params.objects('balloon')) + i
+        try:
+            vm = env.get_vm(vm_name)
+            vm.devices.simple_hotplug(dev, vm.monitor)
+        except QMPCmdError:
+            if dev_num < 1:
+                raise
```

Comment:
Should test fail if dev_num<1 ?

Reply:
@huangyum Do you mean I should explicitly write `test.fail` here? Cause I think raise the error will fail the test case naturally,.

Reply:
Yes. I'm not sure if 'raise' would lead to FAIL or ERROR.